telemetry/specs/telemetry_report.mdk

```diff
@@ -550,158 +653,178 @@ particular deployment.
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
-TLVType (4b): TLV Type – TLV data type.
-0 – Domain Specific Extension Data
-1 - Ethernet
-2 – IPv4
-3 – IPv6
-4-15 – Reserved.
+* Variable Optional Domain Specific Metadata
+  : The metadata corresponding to the *DSMdBits*, 4 octets or a multiple of
+    4 octets for each bit.
+
+### Inner Report Contents for *InType* 1 (*TLV*) (4+ octets) { #tlv }
+
+One or more TLVs, each following the format defined in this section.
+The presence of multiple TLVs can be determined by comparing the
+*TLVLength* in the first TLV with the *Report Length* in the Individual
+Report Header.
+
+```
+   0                   1                   2                   3
+   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+  |TLVType|  Rsvd |   TLVLength   |        TLV Data Template      |
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+  |          Variable TLV Data as identified by TLVType           |
+  |      (Truncated Packet or Domain Specific Extension Data)     |
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+* TLVType (4b): TLV data type
+  :
+    - **0**: Domain Specific Extension Data
+    - **1**: Ethernet
+    - **2**: IPv4
+    - **3**: IPv6
+    - **4-15**:  Reserved.
+
+* Rsvd (4b) – Reserved
+  : Should be set to zero upon transmission and ignored upon reception.
+
+* TLVLength (8b)
+  : Indicates the length, in 4-byte words, of *Variable TLV Data* as identified by *TLVType*.
+
+* TLV Data Template (16b)
+  : Specifies the format of the *Variable TLV Data*.
+    A non-zero *TLV Data Template* value specifies the template for *TLVType*
+    codepoint of *Domain Specific Extension Data*. For *TLVType* codepoints
+    *Ethernet*, *IPv4*, and *IPv6*, the *TLV Data Template* value should be
+    zero upon transmission and ignored upon reception.
+
+* Variable TLV Data
+  : Variable length data based upon *TLVType*. The following two fields are
+    defined in this version.
 
-Rsvd (4b) – Reserved. Should be set to zero upon transmission and ignored upon reception.
+* Truncated Packet
+  : L2/L3/ESP/L4 of the packet for flow details. Presence of this field is
+    indicated by *TLVType* codepoint 1, 2, or 3, which identifies the type of
+    header at the beginning of the truncated packet.
 
-TLVLength (8b) - Indicates the length, in 4-byte word, of Variable TLV Data as identified by TLVType.
+* Domain Specific Extension Data
+  : Domain Specific Extension Data, whose format can be determined from the
+    *Domain Specific ID* specified in the *Telemetry Report Contents* and the
+    *TLV Data Template*. For *RepType* codepoint 1 *INT*, this is additional
+    domain specific data, not associated with *DSMdBits*.
+    Presence of this field is indicated by *TLVType* codepoint 0.
```

Comment:
These different combinatory cases are quite confusing.
Wonder if we can add a table to summarize them?